tests/memgraph.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2016-2020 Memgraph Ltd. [https://memgraph.com]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "mgclient.h"
+extern "C" {
+    #include "mgvalue.h"
+}
+
+template <typename T>
+T getEnvOrDefault(const std::string &valueName, const T &defaultValue)
+{
+    const char *charValue = std::getenv(valueName.c_str());
+    if (charValue)
+    {
+        T value;
+        std::stringstream envValueStream(charValue);
+        envValueStream >> value;
+        return value;
+    }
+    return defaultValue;
+}
+
+template <>
+std::string getEnvOrDefault<std::string>(const std::string &valueName, const std::string &defaultValue)
+{
+    const char *value = std::getenv(valueName.c_str());
+    return value ? value : defaultValue;
+}
```

Comment:
Why is this a separate function?
The case is handled correctly with the above template function.

Reply:
Removed the redundant specialization